### PR TITLE
[internal/patch/grpc-go-insecure] Add patch to grpc-go dependency

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Patches `grpc-go` dependency to add `insecure` package.

### Motivation

Since open-telemetry/opentelemetry-collector#4586, the `go.opentelemetry.io/collector` dependency requires the `insecure` grpc-go package. Since we can't bump `grpc-go` due to etcd-io/etcd#12124, and alternative solutions like bumping Kubernetes are not doable either (see kubernetes/kubernetes#106536), we patch grpc-go through a `replace` directive to be able to bump `go.opentelemetry.io/collector` to the latest version.

### Additional Notes

See more detailed explanation on the `README.md` file.

### Possible Drawbacks / Trade-offs

We are patching an important dependency, which is risky. However, we are only adding new code, and this code is copied verbatim from a future version of the dependency.

### Describe how to test/QA your changes

It should be a no-op. Since this new package will only be imported by the OpenTelemetry Collector dependency, we can QA by checking the OTLP gRPC endpoint. Other gRPC features need no testing.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -35,6 +35,9 @@ replace (
 	google.golang.org/grpc => github.com/grpc/grpc-go v1.28.0
 )
 
+// HACK: Add `insecure` package (added on grpc-go v1.34) to support packages using it (notably go.opentelemetry/collector)
```
Comment (albertvaka) [inline go.mod:38]:
```suggestion
// HACK: Add `insecure` package (added on grpc-go v1.34) to support packages using it (notably go.opentelemetry/collector)
// See internal/patch/grpc-go-insecure/README.md for more details.
```

**Review by albertvaka — APPROVED** [review]

File: internal/patch/grpc-go-insecure/go.mod [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module google.golang.org/grpc/credentials/insecure
+
+go 1.16
+
+// Should match the version we replace on the main go.mod
```
Comment (olivielpeau) [inline internal/patch/grpc-go-insecure/go.mod:5]: should this comment also be mentioned in the top-level go.mod of the repo (next to the line that pins `google.golang.org/grpc`?)

**Review by olivielpeau — APPROVED** [review]
Looks good to me, thanks for adding detailed docs on the change. Just a suggestion on an additional comment.

I think we can mention explicitly in the description that this change doesn't require any specific QA on other parts of the agent that use grpc (other than the otel pipeline), given the nature of the change.

File: internal/patch/grpc-go-insecure/go.mod (hunk 2 above)
Comment (mx-psi) [reply internal/patch/grpc-go-insecure/go.mod:5]: Makes sense, I added a comment

**Comment by mx-psi** [comment]
Updated the QA instructions to mention only OTLP

**Review by albertvaka — APPROVED** [review]

File: internal/patch/grpc-go-insecure/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# `google.golang.org/grpc/credentials/insecure` patch
+
+## What is this?
```
Comment (truthbk) [inline internal/patch/grpc-go-insecure/README.md:3]: 🍰

**Review by truthbk — APPROVED** [review]
Amazing, thank you.

**Pull request merged**